src/nfd_portal.cpp

```diff
@@ -14,8 +14,17 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/random.h>  // for the random token string
-#include <unistd.h>      // for access()
+#include <unistd.h>  // for access()
+
+#if !defined(__has_include) || !defined(__linux__)
+#include <sys/random.h>  // for getrandom() - the random token string
+#elif __has_include(<sys/random.h>)
+#include <sys/random.h>
+#else  // for GLIBC < 2.25
+#include <linux/random.h>
```

Comment:
Why is `<linux/random.h>` required?

Reply:
I only included it for the flag definitions (`GRND_RANDOM`, etc.)

Those flags aren't actually used in the `getrandom` call below, but I was thinking to include them in case they are ever used.

Reply:
I see.  Let's not include headers that are currently unused; we can add it in the future if needed.